File: 3163-string-compression-iii/3163-string-compression-iii.go
func compressedString(word string) string {
    var currCount = 1
    var res = []string{}
    for i := range len(word) - 1 {
        if word[i] != word[i + 1] || currCount == 9 {
            res = append(res, strconv.Itoa(currCount))
            res = append(res, string(word[i]))
            currCount = 1
        } else {
            currCount++
        }
    }
    res = append(res, strconv.Itoa(currCount))
    res = append(res, string(word[len(word) - 1]))
    return strings.Join(res, "")
}

File: 3163-string-compression-iii/3163-string-compression-iii.py
class Solution:
    def compressedString(self, word: str) -> str:
        curr = 1
        ch = word[0]
        res = []
        for c in word[1:] + '-':
            if ch != c or curr == 9:
                res.append(f'{curr}{ch}')
                ch = c
                curr = 0
            
            curr += 1
        
        return ''.join(res)
            